### PR TITLE
Add docs for JobMutableNodeSchedulingDirectives

### PR DESCRIPTION
**Pull request by ahg-g**
Ref https://github.com/kubernetes/enhancements/issues/2926

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 1]
```diff
@@ -610,6 +610,31 @@ controller is tracking a Job using Pod finalizers by checking if the Job has the
 annotation `batch.kubernetes.io/job-tracking`. You should **not** manually add
 or remove this annotation from Jobs.
 
+### Mutable Scheduling Directives
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:613]: Better have this right after the "Suspending a Job" section, or even as a subsection within it.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 2]
```diff
@@ -610,6 +610,31 @@ controller is tracking a Job using Pod finalizers by checking if the Job has the
...
+In order to use this behavior, you must enable the `JobMutableNodeSchedulingDirectives`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/).
+It is enabled by default.
+{{< /note >}}
+
+In most cases a parallel job will want the pods to run within specific constraints, 
+like all in the same zone, or all either on GPU model x or y, but not a mix of both.
+
+The Suspend flag is the first step towards achieving those semantics. Suspend allows a
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:627]: I think the documentation for suspend is not very explicit in telling that you can create jobs in a suspended state. Can you fix that?

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 1 above)
Comment (ahg-g) [reply content/en/docs/concepts/workloads/controllers/job.md]: moved it right after the suspend section

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 2 above)
Comment (ahg-g) [reply content/en/docs/concepts/workloads/controllers/job.md]: I added a sentence at the beginning, you think it is enough?

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 3]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+In order to use this behavior, you must enable the `JobMutableNodeSchedulingDirectives`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/).
+It is enabled by default.
+{{< /note >}}
+
+In most cases a parallel job will want the pods to run within specific constraints, 
+like all in the same zone, or all either on GPU model x or y but not a mix of both.
+
+The Suspend flag is the first step towards achieving those semantics. Suspend allows a
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:523]: s/flag/field

also link to `#suspending-a-job`

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 4]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+
+The Suspend flag is the first step towards achieving those semantics. Suspend allows a 
+custom queue controller to decide when a job should start; However, once a job is unsuspended,
+a custom queue controller has no influence on where the pods of a job will actually land.
+
+This feature allows updating a Job's scheduling bits before it starts, which gives custom queue
+controllers the ability to influence pod placement while at the same time offloading actual 
+pod-to-node assignment to the existing kube-scheduler.
+
+Specifically, when this feature is enabled, node affinity, node selector, tolerations,
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:531]: Probably no need to mention the feature gate again here.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 5]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+In most cases a parallel job will want the pods to run within specific constraints, 
+like all in the same zone, or all either on GPU model x or y but not a mix of both.
+
+The Suspend flag is the first step towards achieving those semantics. Suspend allows a 
+custom queue controller to decide when a job should start; However, once a job is unsuspended,
+a custom queue controller has no influence on where the pods of a job will actually land.
+
+This feature allows updating a Job's scheduling bits before it starts, which gives custom queue
+controllers the ability to influence pod placement while at the same time offloading actual 
+pod-to-node assignment to the existing kube-scheduler.
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:529]: to kube-scheduler.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 6]
```diff
@@ -794,6 +795,10 @@ Each feature gate is designed for enabling/disabling a specific feature:
   Initializers admission plugin.
 - `IPv6DualStack`: Enable [dual stack](/docs/concepts/services-networking/dual-stack/)
   support for IPv6.
+- `JobMutableNodeSchedulingDirectives`: Allows updating node scheduling directives in
+  the pod template of [Job](/docs/concepts/workloads/controllers/job).
+  Specifically, node affinity, selector, tolerations, labels and annotations of the pod template.
```
Comment (alculquicondor) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:800]: I think these details can be left out

Comment (ahg-g) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: removed.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 7]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+In order to use this behavior, you must enable the `JobMutableNodeSchedulingDirectives`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/).
+It is enabled by default.
+{{< /note >}}
+
+In most cases a parallel job will want the pods to run within specific constraints, 
+like all in the same zone, or all either on GPU model x or y but not a mix of both.
+
+The [Suspend](#suspending-a-job) field is the first step towards achieving those semantics. Suspend allows a
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:523]: s/Suspend/suspend

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 8]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+{{< /note >}}
+
+In most cases a parallel job will want the pods to run within specific constraints, 
+like all in the same zone, or all either on GPU model x or y but not a mix of both.
+
+The [Suspend](#suspending-a-job) field is the first step towards achieving those semantics. Suspend allows a 
+custom queue controller to decide when a job should start; However, once a job is unsuspended,
+a custom queue controller has no influence on where the pods of a job will actually land.
+
+This feature allows updating a Job's scheduling bits before it starts, which gives custom queue
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/controllers/job.md:527]: s/bits/directives

**Comment by ahg-g** [comment]
done.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 9]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
+{{< feature-state for_k8s_version="v1.23" state="beta" >}}
+
+{{< note >}}
+In order to use this behavior, you must enable the `JobMutableNodeSchedulingDirectives`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/).
+It is enabled by default.
+{{< /note >}}
+
+In most cases a parallel job will want the pods to run within specific constraints,
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/controllers/job.md:520]: nit: "with constraints" or "within constraints"?

Comment (ahg-g) [reply content/en/docs/concepts/workloads/controllers/job.md]: done.

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 10]
```diff
@@ -503,6 +506,32 @@ directly a result of toggling the `.spec.suspend` field. In the time between
...
 
+### Mutable Scheduling Directives
+
+{{< feature-state for_k8s_version="v1.23" state="beta" >}}
+
+{{< note >}}
+In order to use this behavior, you must enable the `JobMutableNodeSchedulingDirectives`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/).
+It is enabled by default.
```
Comment (soltysh) [inline content/en/docs/concepts/workloads/controllers/job.md:517]: Nit: this reads weird, maybe change the last sentence to say:


```suggestion
In v1.23 it is enabled by default. 
```

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**